feign-hystrix-opentracing/src/main/java/feign/opentracing/hystrix/TracingConcurrencyStrategy.java

```diff
@@ -60,33 +65,37 @@ private TracingConcurrencyStrategy(Tracer tracer) {
         }
 
         Callable<T> delegateCallable = this.delegateStrategy == null ? callable :
-                this.delegateStrategy.wrapCallable(callable);
+            this.delegateStrategy.wrapCallable(callable);
 
         if (delegateCallable instanceof OpenTracingHystrixCallable) {
             return delegateCallable;
         }
 
-        return new OpenTracingHystrixCallable<>(delegateCallable);
+        if (tracer.scopeManager().active() == null) {
+            return delegateCallable;
+        }
+
+        return new OpenTracingHystrixCallable<>(delegateCallable, tracer.scopeManager(), tracer.scopeManager().active().span());
     }
 
-    private class OpenTracingHystrixCallable<S> implements Callable<S> {
-        private Callable<S> delegateCallable;
-        private ActiveSpan parentSpan;
+    private static class OpenTracingHystrixCallable<S> implements Callable<S> {
+        private final Callable<S> delegateCallable;
+        private ScopeManager scopeManager;
+        private Span span;
 
-        public OpenTracingHystrixCallable(Callable<S> delegate) {
+        public OpenTracingHystrixCallable(Callable<S> delegate, ScopeManager scopeManager, Span span) {
+            if (span == null || delegate == null || scopeManager == null) {
+                throw new NullPointerException();
```

Comment:
Be nice to the caller: tell them *which* one is null :)

Reply:
Is it necessary this is an internal private class. Checks are also done before calling the constructor.